misc/test_expected_payload_cooker.py

```diff
@@ -0,0 +1,197 @@
+    def attempt_decrypt(raw: bytes) -> bytes | None:
+        for key in KEY_CANDIDATES:
+            key = nacl.secret.SecretBox(key)
+            try:
+                return key.decrypt(raw)
+            except nacl.exceptions.CryptoError:
+                continue
+
+    def attempt_decompression(raw: bytes) -> bytes | None:
+        try:
+            return zstandard.decompress(raw)
+        except ValueError:
+            return None
+
+    def attempt_deserialization(raw: bytes) -> dict[str, object] | None:
+        if raw[0] == 0xFF:
+            try:
+                return msgpack.unpackb(raw[1:])
+            except ValueError:
+                pass
+
+        # First byte is the version
+        if raw[0] != 0x00:
+            return None
+
+        try:
+            decompressed = zstandard.decompress(raw[1:])
+        except ValueError:
+            return None
+
+        try:
+            return msgpack.unpackb(decompressed)
+        except ValueError:
+            return None
+
+    # First attempt: consider the data is not signed nor encrypted
+    deserialized = attempt_deserialization(raw)
+    if deserialized is not None:
+        return deserialized
+
+    # Second attempt: consider the data is only signed
+    raw_without_signature = raw[64:]
+    deserialized = attempt_deserialization(raw_without_signature)
+    if deserialized is not None:
+        return deserialized
+
+    # Last attempt: consider the data is signed and encrypted
+    decrypted = attempt_decrypt(raw)
+    assert decrypted is not None
+    decrypted_without_signature = decrypted[64:]
+    decompressed = attempt_decompression(decrypted_without_signature)
+    assert decompressed is not None
+    deserialized = attempt_deserialization(decompressed)
+    assert deserialized is not None
+    return deserialized
+
+
+def cook_msgpack_type(value):
+    if isinstance(value, bytes):
+        return value.hex()
+
+    if isinstance(value, msgpack.ExtType):
+        match value.code:
+            # DateTime
+            case 1:
+                ts = struct.unpack(">q", value.data)[0]
+                dt = datetime.datetime.fromtimestamp(ts / 1000000).isoformat() + "Z"
+                return f"ext(1, {ts}) i.e. {dt}"
+            # UUID
+            case 2:
+                return f"ext(2, {value.data.hex()})"
+
+            case _:
+                pass
+
+    return value
+
+
+current_test = None
+
+lines = sys.stdin.readlines()
+
+for line in lines:
+    m = FAILING_TEST_HEADER_PATTERN.match(line)
+    if m:
+        current_test = m.group(1)
+
+    m = TAG_PATTERN.match(line)
+    if not m:
+        continue
+
+    expected_raw = bytes(bytearray([int(c.strip()) for c in m.group(1).split(",")]))
+
+    # TODO: decode does not seem to work for all tests
+    try:
+        expected_decoded = decode_expected_raw(expected_raw)
+    except Exception:
+        expected_decoded = None
+
+    txt = []
+    txt.append("    // Generated from Parsec v3.0.0-b.11+dev")
```

Comment:
Maybe get the version from `Cargo.toml`?

Reply:
Done by reading the content of `libparsec/version`.

However it currently contains `v3.0.0-b.6+dev` instead of `v3.0.0-b.11+dev`. Did we forget any post-release step?

Reply:
You mean the same value contained in:

https://github.com/Scille/parsec-cloud/blob/7c3cf393e74df5b449cb7e2f5ffc8379a44e3b99/Cargo.toml#L57

or

https://github.com/Scille/parsec-cloud/blob/7c3cf393e74df5b449cb7e2f5ffc8379a44e3b99/misc/version_updater.py#L176

? :stuck_out_tongue: 

We never acknowledge the 3.0.0.b.11